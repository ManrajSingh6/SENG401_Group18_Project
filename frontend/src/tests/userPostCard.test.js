import React from 'react';
import renderer from 'react-test-renderer';
import PostCard from '../components/userPostCard';
import '@testing-library/jest-dom';
import { render, screen, fireEvent,cleanup} from '@testing-library/react';
import ThumbUpRoundedIcon from '@mui/icons-material/ThumbUpRounded';
import MessageRoundedIcon from '@mui/icons-material/MessageRounded';


// testing delete button
jest.mock('../components/userPostCard', () => {
    const postCard = () => (
      <div data-testid='postCard'>user post card</div>
    );
  
    return postCard;
  });
  
  
it('Confirm deletion button', ()=>{
render(<PostCard/>);
const button = screen.getByTestId('postCard', {name: "button"});
fireEvent.click(button);
expect(screen.getByTestId("postCard")).toBeInTheDocument();
});

//testing like icon
test('renders ThumbUpRoundedIcon', () => {
    const testme = renderer.create(<ThumbUpRoundedIcon />).toJSON();
    expect(testme).toMatchSnapshot();
  });
  
  // testing comment icon
  test('renders MessageRoundedIcon', () => {
      const testme = renderer.create(<MessageRoundedIcon />).toJSON();
      expect(testme).toMatchSnapshot();
    });