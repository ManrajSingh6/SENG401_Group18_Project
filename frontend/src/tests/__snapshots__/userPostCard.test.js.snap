// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MessageRoundedIcon 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="MessageRoundedIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-3 12H7c-.55 0-1-.45-1-1s.45-1 1-1h10c.55 0 1 .45 1 1s-.45 1-1 1zm0-3H7c-.55 0-1-.45-1-1s.45-1 1-1h10c.55 0 1 .45 1 1s-.45 1-1 1zm0-3H7c-.55 0-1-.45-1-1s.45-1 1-1h10c.55 0 1 .45 1 1s-.45 1-1 1z"
  />
</svg>
`;

exports[`renders ThumbUpRoundedIcon 1`] = `
<svg
  aria-hidden={true}
  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="ThumbUpRoundedIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M13.12 2.06 7.58 7.6c-.37.37-.58.88-.58 1.41V19c0 1.1.9 2 2 2h9c.8 0 1.52-.48 1.84-1.21l3.26-7.61C23.94 10.2 22.49 8 20.34 8h-5.65l.95-4.58c.1-.5-.05-1.01-.41-1.37-.59-.58-1.53-.58-2.11.01zM3 21c1.1 0 2-.9 2-2v-8c0-1.1-.9-2-2-2s-2 .9-2 2v8c0 1.1.9 2 2 2z"
  />
</svg>
`;
